include/maidsafe/crux/socket.hpp

```diff
@@ -60,8 +60,7 @@ class socket
         typename boost::asio::handler_type<CompletionToken,
                                            void(boost::system::error_code)>::type
         >::type
-    async_connect(const endpoint_type& remote_endpoint,
-                  CompletionToken&& token);
+    async_connect(endpoint_type remote_endpoint, CompletionToken&& token);
```

Comment:
Minor comment: Is there any reason to change this to pass-by-value?

I am actually considering changing all the other `endpoint_type` arguments to pass-by-const-ref to avoid superfluous copying.